### PR TITLE
Make only admin password mandatory.

### PR DESCRIPTION
**Pull request by omron93**
User have to only specify `MONGODB_ADMIN_PASSWORD` and optionally provide settings for user with 'readWrite' role.

Resolves: #32

### CODE REVIEW

**Comment by omron93** [comment]
[test-openshift]

**Comment by omron93** [comment]
[test]

**Comment by omron93** [comment]
Changes applied to all versions. [test-openshift]

@bparees Please review.

File: 2.6/root/usr/share/container-scripts/mongodb/init-replset.sh [hunk 1]
```diff
@@ -43,7 +43,7 @@ function initiate() {
 
   info "Creating MongoDB users ..."
   mongo_create_admin
-  mongo_create_user "-u admin -p ${MONGODB_ADMIN_PASSWORD}"
+  [[ -v CREATE_USER ]] && mongo_create_user "-u admin -p ${MONGODB_ADMIN_PASSWORD}"
```
Comment (bparees) [inline 2.6/root/usr/share/container-scripts/mongodb/init-replset.sh:46]: if the admin password is required, why are you not always creating the admin user here?

File: 2.6/test/run [hunk 2]
```diff
@@ -45,6 +45,42 @@ function mongo_admin_cmd() {
     docker run --rm $IMAGE_NAME mongo admin --host $CONTAINER_IP -u admin -p "$ADMIN_PASS" --eval "${@}"
 }
 
+# Make sure the invocation of docker run fails.
+function assert_container_creation_fails() {
+
+  # Time the docker run command. It should fail. If it doesn't fail,
+  # postgresql will keep running so we kill it with SIGKILL to make sure
```
Comment (bparees) [inline 2.6/test/run]: postgresql?

**Review by bparees — CHANGES_REQUESTED** [review]
what's the motivation for this change?

edit: nm, i see the issue discussion.

File: 2.6/test/run (hunk 2 above)
Comment (omron93) [reply 2.6/test/run]: Sorry, it was copied from postgresql-container. Fixed.

File: 2.6/root/usr/share/container-scripts/mongodb/init-replset.sh (hunk 1 above)
Comment (omron93) [reply 2.6/root/usr/share/container-scripts/mongodb/init-replset.sh:46]: `mongo_create_admin` line creates admin user.

`mongo_create_user` creates "normal" user (passed string contains parameters for `mongo` shell - needs to know how to login)

**Comment by omron93** [comment]
> what's the motivation for this change?
edit: nm, i see the issue discussion.

@bparees  As noted there, other databases also do not require both users [1][2].

In mongodb admin user is not created by default, so it has to be created (otherwise auth have to be disabled and server restarted to allow admin changes). Normal user might not be necessary, also number of env vars needed for running container (for example for testing) is lower :-) 

[1] https://github.com/sclorg/postgresql-container/blob/master/9.5/root/usr/share/container-scripts/postgresql/common.sh#L34
[2] https://github.com/sclorg/mariadb-container/blob/master/10.1/root/usr/share/container-scripts/mysql/validate-variables.sh#L1

[test-openshift]

**Comment by bparees** [comment]
lgtm

**Comment by omron93** [comment]
Squashed and rebased.

**Comment by omron93** [comment]
[test-openshift]

**Comment by omron93** [comment]
[test][test-openshift]

**Comment by omron93** [comment]
LGTM from @bparees and tests are passing. So merging.

**Pull request merged**